### ISSUE TITLE
resolve-taghelper performance issue

### ISSUE THREAD

**Issue by IlyaKhD** [aspnet/RazorTooling#35]
Each time a .cshtml file is reopened in VisualStuido 2015 a new process created:

`Microsoft.Dnx.ApplicationHost Microsoft.AspNet.Tooling.Razor resolve-taghelpers --protocol 1 Microsoft.AspNet.Mvc.Razor <OtherTagHelperAssemblies>`

While it is being executed the Intellisense doesn't work (this brings some annoying inconvenience), and for some reason this process takes much more time for documented TagHelpers.

Check `PerfTest` console app in this [branch](https://github.com/IlyaKhD/RazorTooling/tree/khd/resolve-taghelpers.performance).

I've created two fake TagHelper NuGet packs (hosted on MyGet) containing 200 TagHelpers (each has 20 properties). The only difference is that properties in the second pack have summary.
You can run the following commands:

`> dnx resolve-taghelpers`
`> dnx resolve-taghelpers-withsummary`

On my environment it takes 3 sec and 21 sec accordingly.

**Comment by IlyaKhD** [comment]
The thing is hidden in AspNet/Razor.
Documentation file is read each time a `DocumentationDescriptor` is built, though while resolving `TagHelpers` this file could be loaded for each assembly only once.

The issue is illustrated here: https://github.com/IlyaKhD/Razor/commit/db7d1a5b4860fe3e5ef19bc889cd3f37e0f812c0

This is not a fix, of course, put caching brings me from 21 sec to 5sec, which is 4 times better.

**Comment by NTaylorMullen** [comment]
@IlyaKhD thanks for the investigation! I've looked at your branch and profiled it and it turns out the `System.Xml.Linq.XDocument` calls we're performing are slowing the system down considerably.

Caching is definitely the correct fix for this in Razor! Would you mind filing an issue on aspnet/Razor/issues for this? We'll definitely fix this for RC2.

**Comment by IlyaKhD** [comment]
I've filed a new issue: https://github.com/aspnet/Razor/issues/630

Btw, what I mean by 'this is not a fix' is that there several classes which could be responsible for caching an assembly documentation file, so I'm not quite sure this fix is an appropriate one.

**Comment by NTaylorMullen** [comment]
Thanks @IlyaKhD! Awesome bug/follow up. In the end there's just not enough high volume `TagHelper` uses out there quite yet so what you reported is super valuable :smile: 

Closing since you created a Razor issue.
